pkg/controller/operators/openshift/helpers.go

```diff
@@ -110,7 +110,7 @@ func (s skew) String() string {
 		return fmt.Sprintf("%s/%s has invalid %s properties: %s", s.namespace, s.name, MaxOpenShiftVersionProperty, s.err)
 	}
 
-	return fmt.Sprintf("%s/%s is incompatible with OpenShift versions greater than %s", s.namespace, s.name, s.maxOpenShiftVersion)
+	return fmt.Sprintf("%s/%s is incompatible with OpenShift minor versions greater than %s", s.namespace, s.name, s.maxOpenShiftVersion)
```

Comment:
I believe `skew.maxOpenShiftVersion` is the result of directly stringifying the value of the property, which **can** specify patch versions. I would take care to format the message so that only `major.minor` is displayed.

Reply:
Could the test include a maxOpenShift version that includes a patch and could we ensure that the patch version is omitted from the message?

Reply:
Shouldn't we just throw a validation error on a patch version? It doesn't seem to me to make a lot of sense to allow users to even set it like that if we can't actually block those upgrades anyway.

Reply:
How about requiring the operators to declare `maxOpenShiftVersion: 4.8.9999` if they support all future 4.8.* versions?

Reply:
I just think it's confusing and erroneous to do that. OLM functionally can't do anything special with the patch version here, so we should limit it to the minor version with validation.

Reply:
Modified it to drop the build version and complain on a non-zero patch/pre-release version. I've added a warning on the api validation https://github.com/operator-framework/api/pull/146